docs/deploymentscenarios.adoc

```diff
@@ -0,0 +1,43 @@
+
+[[getting-started-scenario]]
+=== Getting Started Scenario
+
+This scenario is the scenario most users probably want to choose when they are just getting started with JanusGraph. It offers scalability and fault tolerance with a minimum number of servers required. JanusGraph Server runs together with an instance of the storage backend and optionally also an instance of the index backend on every server.
+
+image:getting-started-scenario.svg[Getting started deployment scenario diagram, 650]
+
+A setup like this can be extended by simply adding more servers of the same kind or by moving one of the components onto dedicated servers. The latter describes a growth path to transform the deployment into the <<advanced-scenario,Advanced Scenario>>.
+
+Any of the scalable storage backends can be used with this scenario. Note however that for Scylla http://docs.scylladb.com/getting-started/scylla_in_a_shared_environment/[some configuration is required when it is hosted co-located with other services] like in this scenario. When an index backend should be used in this scenario then it also needs to be one that is scalable.
+
+[[advanced-scenario]]
+=== Advanced Scenario
+
+The advanced scenario is an evolution of the <<getting-started-scenario>>. Instead of hosting the JanusGraph Server instances together with the storage backend and optionally also the index backend, they are now separated on different servers.
+The advantage of hosting the different components (JanusGraph Server, storage/index backend) on different servers is that they can be scaled and managed independently of each other.
+This offers a higher flexibility at the cost of having to maintain more servers.
+
+image:advanced-scenario.svg[Advanced deployment scenario diagram, 800]
+
+Since this scenario offers independent scalability of the different components, it of course makes most sense to also use scalable backends.
+
+[[minimalist-scenario]]
+=== Minimalist Scenario
+
+It is also possible to host JanusGraph Server together with the backend(s) on just one server. This is especially attractive for testing purposes or for example when JanusGraph just supports a single application which can then also run on the same server.
+
+image:minimalist-scenario.svg[Minimalist deployment scenario diagram, 650]
+
+Opposed to the previous scenarios, it makes most sense to use backends for this scenario that are not scalable. The in-memory backend can be used for testing purposes or Berkeley DB for production and Lucene as the optional index backend.
+
+[[embedded-janusgraph]]
+=== Embedded JanusGraph
+
+Instead of connecting to the JanusGraph Server from an application it is also possible to embed JanusGraph as a library inside a JVM based application. While this reduces the administrative overhead it makes it impossible to scale JanusGraph independently of the application.
```

Comment:
missing a comma in here:

While this reduces the administrative overhead, it makes it impossible to scale JanusGraph independently of the application.